src/Native/CMakeLists.txt

```diff
@@ -99,6 +99,114 @@ if (CMAKE_SYSTEM_NAME STREQUAL Linux)
+endif(CMAKE_SYSTEM_NAME STREQUAL SunOS)
+
+if (NOT WIN32)
+    if (CMAKE_SYSTEM_NAME STREQUAL Darwin)
+        # Ensure that dsymutil and strip is present
+        find_program(DSYMUTIL dsymutil)
+        if (DSYMUTIL STREQUAL "DSYMUTIL-NOTFOUND")
+            message(FATAL_ERROR "dsymutil not found")
+        endif()
+        find_program(STRIP strip)
+        if (STRIP STREQUAL "STRIP-NOTFOUND")
+            message(FATAL_ERROR "strip not found")
+        endif()
+    else (CMAKE_SYSTEM_NAME STREQUAL Darwin)
+        # Ensure that objcopy is present
+        find_program(OBJCOPY objcopy)
+        if (OBJCOPY STREQUAL "OBJCOPY-NOTFOUND")
+            message(FATAL_ERROR "objcopy not found")
+        endif()
+    endif (CMAKE_SYSTEM_NAME STREQUAL Darwin)
+endif ()
+
+
+function(strip_symbols targetName outputFilename)
+    if(CLR_CMAKE_PLATFORM_UNIX)
+        if(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE)
+
+            # On the older version of cmake (2.8.12) used on Ubuntu 14.04 the TARGET_FILE
+            # generator expression doesn't work correctly returning the wrong path and on
+            # the newer cmake versions the LOCATION property isn't supported anymore.
+            if(CMAKE_VERSION VERSION_EQUAL 3.0 OR CMAKE_VERSION VERSION_GREATER 3.0)
+                set(strip_source_file $<TARGET_FILE:${targetName}>)
+            else()
+                get_property(strip_source_file TARGET ${targetName} PROPERTY LOCATION)
+            endif()
+
+            if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
+                set(strip_destination_file ${strip_source_file}.dwarf)
+
+                add_custom_command(
+                    TARGET ${targetName}
+                    POST_BUILD
+                    VERBATIM 
+                    COMMAND ${DSYMUTIL} --flat --minimize ${strip_source_file}
+                    COMMAND ${STRIP} -u -r ${strip_source_file}
+                    COMMENT Stripping symbols from ${strip_source_file} into file ${strip_destination_file}
+                )
+            else(CMAKE_SYSTEM_NAME STREQUAL Darwin)
+                set(strip_destination_file ${strip_source_file}.dbg)
+
+                add_custom_command(
+                    TARGET ${targetName}
+                    POST_BUILD
+                    VERBATIM 
+                    COMMAND ${OBJCOPY} --only-keep-debug ${strip_source_file} ${strip_destination_file}
+                    COMMAND ${OBJCOPY} --strip-unneeded ${strip_source_file}
+                    COMMAND ${OBJCOPY} --add-gnu-debuglink=${strip_destination_file} ${strip_source_file}
+                    COMMENT Stripping symbols from ${strip_source_file} into file ${strip_destination_file}
+                )
+            endif(CMAKE_SYSTEM_NAME STREQUAL Darwin)
+
+            set(${outputFilename} ${strip_destination_file} PARENT_SCOPE)
+        endif(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE)
+    endif(CLR_CMAKE_PLATFORM_UNIX)
+endfunction()
+
+function(install_clr targetName)
```

Comment:
NIT: I believe I understand why this function is named like it is (since these targets initially came from CoreCLR), but maybe we should instead call it `install_corefx` or `install_library_and_symbols` ?